src/Tickets/Commerce/Models/Order_Model.php

```diff
@@ -179,4 +171,40 @@ protected function get_cache_slug() {
 		return 'tc_orders';
 	}
 
+	/**
+	 * Returns a callback that will return the ticket data for the order items.
+	 *
+	 * @since TBD
+	 *
+	 * @param $items array The order items.
+	 *
+	 * @return \Closure
+	 */
+	public function get_ticket_data_callback( $items ): \Closure {
+		/**
+		 * Filters the callback that will return the ticket data for the order items.
+		 *
+		 * Returning a non `null` value here will skip the default logic.
+		 *
+		 * @since TBD
+		 *
+		 * @param null|\Closure $callback The callback that will return the ticket data for the order items.
+		 * @param array $items The order items.
+		 */
+		$callback = apply_filters( 'tec_tickets_order_items_ticket_data_callback', null, $items );
+
+		if ( null !== $callback ) {
+			return $callback;
+		}
+
+		return static function() use ( $items ) {
+			return array_reduce( $items, function( $tickets, $item ) {
+				// @todo @rafsuntaskin should be updated later to make use of `tec_tc_get_ticket` function, once Ticket Model is updated.
+				$post = get_post( $item['ticket_id'] );
+				$post->ticket_data = $item;
+				$tickets[] = $post;
+				return $tickets;
+			}, []);
+		};
```

Comment:
Should we pass the default filter via this filter? Allowing someone to actually apply that callback and return another?

Reply:
I have taken this pattern from TEC. I assume this is done this way because if someone is going to apply their own callback then they will iterate over the items again anyway. So, skipping our callback saves that extra iteration which might be unnecessary in case of overrides.